### PR TITLE
fix(dav): Ensure share properties are also set on public remote endpoint

### PR DESCRIPTION
**Pull request by susnux**
## Summary

Ensure that public and internal endpoint both provide the necessary data, so we can e.g. on both request `nc:note`.
As the share note is not something only for users but also for public shares.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]
This is on the very edge between a fix and a feat... 😅🙈
Approved nonetheless!

File: apps/dav/lib/Connector/Sabre/Node.php [hunk 1]
```diff
@@ -298,15 +299,14 @@ public function getSharePermissions($user) {
...
-		} catch (StorageNotAvailableException $e) {
-			$storage = null;
+			$storage = $this->node->getStorage();
+		} catch (NotFoundException $e) {
+			return [];
 		}
 
-		if ($storage && $storage->instanceOfStorage(\OCA\Files_Sharing\SharedStorage::class)) {
+		$attributes = [];
+		if (method_exists($storage, 'getShare')) {
```
Comment (come-nc) [inline apps/dav/lib/Connector/Sabre/Node.php:309]: Why the change from `instanceOfStorage` to `method_exists`?
Seems way more fragile.

Comment (come-nc) [reply apps/dav/lib/Connector/Sabre/Node.php:309]: Hum ok I see that you’ve added the same method to PublicShareWrapper. Would it be possible to add an interface implemented by both classes and use that in instanceOfStorage?

File: apps/dav/lib/Connector/Sabre/Node.php [hunk 2]
```diff
@@ -319,29 +319,25 @@ public function getShareAttributes(): array {
...
 		}
 
-		// Retrieve note from the share object already loaded into
-		// memory, to avoid additional database queries.
-		$storage = $this->getNode()->getStorage();
-		if (!$storage->instanceOfStorage(\OCA\Files_Sharing\SharedStorage::class)) {
+		if (!method_exists($storage, 'getShare')) {
 			return '';
 		}
 		/** @var \OCA\Files_Sharing\SharedStorage $storage */
```
Comment (come-nc) [inline apps/dav/lib/Connector/Sabre/Node.php:332]: This psalm typing is not true anymore

Comment (susnux) [reply apps/dav/lib/Connector/Sabre/Node.php:332]: See your other comment it helps Psalm to understand there is that method ;)
(dirty maybe need to fix your other comment).

File: apps/dav/lib/Connector/Sabre/Node.php (hunk 1 above)
Comment (susnux) [reply apps/dav/lib/Connector/Sabre/Node.php:309]: Problem is the `SharedStorage` is only for internal shares so public shares are not an instance of it.

> Would it be possible to add an interface implemented by both classes and use that in instanceOfStorage?

Like having `IShareStorage` with that method and just check for `instanceOfStorage(IShareStorage)` ?

Comment (come-nc) [reply apps/dav/lib/Connector/Sabre/Node.php:309]: Yes.
Otherwise that will bite us later.

File: apps/files_sharing/lib/SharedStorage.php [hunk 3]
```diff
@@ -35,7 +36,7 @@
 /**
  * Convert target path to source path and pass the function call to the correct storage provider
  */
-class SharedStorage extends \OC\Files\Storage\Wrapper\Jail implements ISharedStorage, IDisableEncryptionStorage {
+class SharedStorage extends \OC\Files\Storage\Wrapper\Jail implements LegacyISharedStorage, ISharedStorage, IDisableEncryptionStorage {
```
[the remark this answers is not in the document]
Comment (susnux) [reply apps/files_sharing/lib/SharedStorage.php:39]: Deprecated on purpose but we need it here for backwards compatibility, remove with Nextcloud 34.

**Review by miaulalala — APPROVED** [review]
